extensions/spark/kyuubi-spark-lineage/src/main/scala/org/apache/kyuubi/plugin/lineage/helper/SparkSQLLineageParseHelper.scala

```diff
@@ -160,15 +162,22 @@ trait LineageParser {
     }
   }
 
+  private def isNameWithQualifier(attr: Attribute, qualifier: Seq[String]): Boolean = {
+    val nameTokens = attr.name.split('.')
+    val namespace = nameTokens.init.mkString(".")
+    nameTokens.length > 1 && namespace.endsWith(qualifier.mkString("."))
+  }
+
   private def mergeRelationColumnLineage(
       parentColumnsLineage: AttributeMap[AttributeSet],
       relationOutput: Seq[Attribute],
       relationColumnLineage: AttributeMap[AttributeSet]): AttributeMap[AttributeSet] = {
     val mergedRelationColumnLineage = {
-      relationOutput.foldLeft((ListMap[Attribute, AttributeSet](), relationColumnLineage)) {
-        case ((acc, x), attr) =>
-          (acc + (attr -> x.head._2), x.tail)
-      }._1
+      relationOutput.slice(0, relationColumnLineage.size)
```

Comment:
should the length of `relationColumnLineage` always same with `relationOutput` ?

Reply:
this case:
```
cache table c2 select * from (
     select b, a, row_number() over (partition by a order by b asc ) rank from t2)
      where rank=1
```
the logicalPlan of `CacheTable`'s subquery is 
```
Filter (isnotnull(rank#4) AND (rank#4 = 1))
+- Window [row_number() windowspecdefinition(a#9, b#10 ASC NULLS FIRST, specifiedwindowframe(RowFrame, unboundedpreceding$(), currentrow$())) AS rank#4], [a#9], [b#10 ASC NULLS FIRST]
   +- Project [b#10, a#9]
      +- HiveTableRelation [`default`.`t2`, org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe, Data Cols: [a#9, b#10], Partition Cols: []]
```
the output of `CacheTable` is [#a, #b, #rank], but the subquery's lineage of `CacheTable` is not include `#rank` column

Reply:
> subquery's lineage of CacheTable is not include #rank column

why does the subquery's lineage of CacheTable is not include `#rank` column ? should it return `rank -> [a, b]` ? or do you mean we do not support collect lineage from window ?

Reply:
Yes, we do not currently have` window`-specific parsing.  I'm not sure if `#rank` needs to be mapped to `[a, b]` here either

Reply:
Can we add the Window pattern here to only parse the lineage of child output and add some comments to explain we now do not support window function ? then I think we do not need this `slice` ?